compiler/syntax/src/res_printer.ml

```diff
@@ -4403,23 +4401,28 @@ and print_jsx_expression ~state lident args cmt_tbl =
               ]);
        ])
 
-and print_jsx_fragment ~state expr cmt_tbl =
+and print_jsx_fragment ~state (children : Parsetree.expression list) cmt_tbl =
   let opening = Doc.text "<>" in
   let closing = Doc.text "</>" in
-  let line_sep =
-    if has_nested_jsx_or_more_than_one_child expr then Doc.hard_line
-    else Doc.line
-  in
+  let line_sep = if List.length children > 0 then Doc.hard_line else Doc.line in
   Doc.group
     (Doc.concat
        [
+         line_sep;
```

Comment:
@shulhi this part isn't 100% correct.
I'm not familiar enough with this part of the codebase.
```res
let z1 = <> <SectionHeader> {React.string("abc")} </SectionHeader> </>
```
will be formatted to
```res

let z1 =
<>
  <SectionHeader> {React.string("abc")} </SectionHeader>
</>
```

so I'm missing some indent here. Not sure how that part works.
Would love to hear your thoughts on this.

Reply:
I checked, the indentation is handled here,

https://github.com/rescript-lang/rescript/blob/9da9b196fcad5ffc53b79e331dbbd4891d6299a6/compiler/syntax/src/res_printer.ml#L2085-L2103

You might need to handle the case for `Pexp_jsx_fragment` here.

Reply:
Thanks, the old code will instead have matched `ParsetreeViewer.has_attributes e.pexp_attributes`. Indentation now works as before.